File: src/api/llm.ts
import { invoke } from "@tauri-apps/api/core";
import type { LlmTiersConfig, LlmTierConfig, TierLevel, TierUsageStats } from "../types";

export async function llmGetTiers(): Promise<LlmTiersConfig> {
  return invoke("llm_get_tiers");
}

export async function llmSetTier(level: TierLevel, config: LlmTierConfig): Promise<void> {
  return invoke("llm_set_tier", { level, config });
}

export async function llmResolveTier(scene: string): Promise<LlmTierConfig> {
  return invoke("llm_resolve_tier", { scene });
}

export async function llmGetUsage(): Promise<TierUsageStats> {
  return invoke("llm_get_usage");
}

export async function llmRecordUsage(tokens: number): Promise<void> {
  return invoke("llm_record_usage", { tokens });
}

export async function llmBindScene(scene: string, level: TierLevel): Promise<void> {
  return invoke("llm_bind_scene", { scene, level });
}

File: src-tauri/src/commands/llm.rs
use tauri::State;

use crate::error::AppResult;
use crate::llm_tiers::{LlmTierConfig, TierLevel, TierUsageStats};
use crate::state::AppState;

#[tauri::command]
pub async fn llm_get_tiers(state: State<'_, AppState>) -> AppResult<serde_json::Value> {
    let mgr = state.llm_tiers.read().await;
    Ok(serde_json::json!({
        "tiers": mgr.tiers,
        "scenes": mgr.scenes,
        "budget": mgr.budget,
    }))
}

#[tauri::command]
pub async fn llm_set_tier(
    state: State<'_, AppState>,
    level: TierLevel,
    config: LlmTierConfig,
) -> AppResult<()> {
    let mut mgr = state.llm_tiers.write().await;
    mgr.tiers.insert(level, config);
    Ok(())
}

#[tauri::command]
pub async fn llm_resolve_tier(
    state: State<'_, AppState>,
    scene: String,
) -> AppResult<LlmTierConfig> {
    let mgr = state.llm_tiers.read().await;
    Ok(mgr.resolve_tier(Some(&scene)).clone())
}

#[tauri::command]
pub async fn llm_get_usage(state: State<'_, AppState>) -> AppResult<TierUsageStats> {
    let mgr = state.llm_tiers.read().await;
    Ok(mgr.get_usage_stats())
}

#[tauri::command]
pub async fn llm_record_usage(state: State<'_, AppState>, tokens: u64) -> AppResult<()> {
    let mgr = state.llm_tiers.read().await;
    mgr.record_usage(tokens);
    Ok(())
}

#[tauri::command]
pub async fn llm_bind_scene(
    state: State<'_, AppState>,
    scene: String,
    level: TierLevel,
) -> AppResult<()> {
    let mut mgr = state.llm_tiers.write().await;
    mgr.scenes.insert(scene, level);
    Ok(())
}
